pkg/controller/manager/manager_controller.go

```diff
@@ -215,6 +216,17 @@ func (r *ReconcileManager) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, nil
 	}
 
+	// check that prometheus is running
+	ns := &corev1.Namespace{}
```

Comment:
I think we should add a watch for this too https://github.com/tigera/operator/pull/475/files#diff-8962c60cf9c10aa1ac7a2ffb1422ae3cL68
So if it is missing and then gets added the Reconcile will be triggered.